src/Common.cpp

```diff
@@ -22,6 +24,29 @@ static const char* getTerminatingSignalErrorMessage(int terminatingSignal) {
   }
 }
 
+std::ostream& operator<<(std::ostream& os, StreamCompletionSignal signal) {
+  switch (signal) {
+    case StreamCompletionSignal::GRACEFUL:
+      return os << "GRACEFUL";
+    case StreamCompletionSignal::ERROR:
+      return os << "ERROR";
+    case StreamCompletionSignal::INVALID_SETUP:
+      return os << "INVALID_SETUP";
+    case StreamCompletionSignal::UNSUPPORTED_SETUP:
+      return os << "UNSUPPORTED_SETUP";
+    case StreamCompletionSignal::REJECTED_SETUP:
+      return os << "REJECTED_SETUP";
+    case StreamCompletionSignal::CONNECTION_ERROR:
+      return os << "CONNECTION_ERROR";
+    case StreamCompletionSignal::CONNECTION_END:
+      return os << "CONNECTION_END";
+    case StreamCompletionSignal::SOCKET_CLOSED:
+      return os << "SOCKET_CLOSED";
+  }
+  // this should be never hit because the switch is over all cases
+  LOG(FATAL) << "unknown StreamCompletionSignal=" << (int)signal;
```

Comment:
Are these meant to be https://github.com/ReactiveSocket/reactivesocket/blob/master/Protocol.md#error-codes or just similarly named but internal only?